torch/linalg/__init__.py

```diff
@@ -88,6 +88,43 @@
 Alias of :func:`torch.det`.
 """)
 
+slogdet = _add_docstr(_linalg.linalg_slogdet, r"""
+linalg.slogdet(input) -> (Tensor, Tensor)
+
+Calculates the sign and natural logarithm of the absolute values of the determinants of the determinant(s)
+of a square matrix or batches of square matrices :attr:`input`.
+
+Supports input of ``float``, ``double``, ``cfloat`` and ``cdouble`` datatypes.
+
+.. note:: When given inputs on a CUDA device, this function synchronizes that device with the CPU.
+
+.. note:: For matrices that have zero determinant, this returns ``(0, -inf)``.
+          If :attr:`input` is batched then the entries in the result tensors corresponding to matrices with
+          the zero determinant have sign 0 and the natural logarithm of the absolute value of the determinant -inf.
+
+Arguments:
+    input (Tensor): the input matrix of size :math:`(n, n)` or the batch of matrices of size :math:`(*, n, n)`
+                    where `*` is one or more batch dimensions.
+
+Returns:
+    A namedtuple (sign, logabsdet) containing the sign of the determinant, and the natural logarithm
```

Comment:
`logabsdet` is a more accurate name than `logdet`, but `logdet` is the name NumPy uses. 

As much as I prefer `logabsdet`, seems like we should use `logdet` for consistency?

Reply:
NumPy doesn't use named tuples and `logdet` is only in its documentation, which could be improved. `logabsdet` is more accurate and I think we should keep it.

Reply:
Good point; I agree with you.